### DISCUSSION TITLE
Method to test Emits (Refreshes)

### DISCUSSION THREAD

**Discussion by iBotPeaches** [livewire/livewire#4298] [3. Feature ideas]
I've got a component that basically just refreshes other components at specific criteria. Looks kinda like

```php
$this->emitTo(OtherLivewireComponent::class, '$refresh');
```

The problem being I have many components that do this pattern, so assertions are presently like:
```php
->assertEmitted('$refresh')
```

I have no support to isolate that assertion to a specific class so can't be 100% sure its refreshing the right one. This is because the object being compared works off the event and props. So this is what I have after digging into the `assertEmitted` class.

```php
array (
  'event' => '$refresh',
  'params' => 
  array (
  ),
  'to' => 'overview-page',
)
```

That `to` property, I'd love to assert again since `$refresh` is too generic for all the components that use this part. Am I doing this wrong or would this be a good enhancement for asserting emits?

I've tried:
```php
->assertEmitted('$refresh', ['to' => 'overview-page']);
->assertEmitted('$refresh', 'overview-page');
```

Until I realized that's not how it works. Thanks!

**Comment by jonneroelofs** [comment]
I think it would be great if we could have either a ->assertEmittedTo() method or could receive the payload in the callback that you can pass to the ->assertEmmited() assertion. Maybe even both.

**Reply by joshhanley** [reply]
@jonneroelofs Yeah I like the idea of a `assertEmittedTo` method and maybe an `assertEmittedUp`?

**Reply by jonneroelofs** [reply]
@joshhanley Did it:
https://github.com/livewire/livewire/pull/4336

**Reply by joshhanley** [reply]
@jonneroelofs thanks! For future reference, please submit PR's that are not on your master branch, maintainers can't contribute to PR's unless they are submitted on feature branches 🙂

**Reply by jonneroelofs** [reply]
Will do
